### ISSUE TITLE
swizzles_avx fails with GCC 5.1

### ISSUE THREAD

**Issue by mkret2** [VcDevel/Vc#9]
https://cdash.gsi.de/testDetails.php?test=272280&build=51127
```
 FAIL: ┍ at /home/mkretz/.Vc-Test/Vc-0.7/tests/swizzles.cpp:115 (0x40da5b):
 FAIL: │ test.badc() ([23591, -7970, 18436, -13100, 29690, 24409, 12135, 29166]) == scalarSwizzle(test, BADC) ([23591, -7970, 18436, -13100, 29690, -7970, 12135, 29166]) -> m[1111 1011] 
 FAIL: ┕ testSwizzle<short_v>
 FAIL: ┍ at /home/mkretz/.Vc-Test/Vc-0.7/tests/swizzles.cpp:115 (0x40f367):
 FAIL: │ test.badc() ([57770, 9820, 42369, 36174, 26448, 46473, 35535, 47990]) == scalarSwizzle(test, BADC) ([57770, 9820, 42369, 36174, 26448, 9820, 35535, 47990]) -> m[1111 1011] 
 FAIL: ┕ testSwizzle<ushort_v>
```

**Comment by mkret2** [comment]
The following indicates that the Vc function works correctly, but the reference value is wrong.
```
Vc:  [23591, -7970, 18436, -13100, 29690, 24409, 12135, 29166]
Ref: [23591, -7970, 18436, -13100, 29690, -7970, 12135, 29166]
Orig:[-7970, 23591, -13100, 18436, 24409, 29690, 29166, 12135]
```

**Comment by mkret2** [comment]
GCC 5.1.0 miscompiles the reference function, which seems to indicate that `AVX::Vector<short>::operator[]` is broken.

**Comment by mkret2** [comment]
This is the relevant bug in GCC: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66866

**Comment by mkret2** [comment]
A possible workaround for Vc is a store to `(u)short[8]` and subsequent array dereference in `Vector<T>::operator[](size_t) const`. I'll wait for the feedback on the GCC issue first, though.

**Comment by mkret2** [comment]
This bug is fixed in GCC on all branches starting from 4.9. I implemented a workaround for the compilers that fail.
